Domain/Shipping/Cargo/Itinerary.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Domain.Shipping.Location;
+using Domain.Shipping.Voyage;
+
+namespace Domain.Shipping.Cargo
+{
+    public class Itinerary 
+    {
+        public IReadOnlyCollection<Leg> Legs { get; }
+
+        public Itinerary(IList<Leg> legs)
+        {
+            if (legs == null)
+                throw new ArgumentNullException(nameof(legs));
+            if (legs.Count == 0)
+                throw new ArgumentException("legs cannot be empty");
+
+            Legs = new ReadOnlyCollection<Leg>(legs);
+        }
+
+        public UnLocode FirstLoadLocation => Legs.First().LoadLocation;
+
+        public VoyageNumber FirstYoyage => Legs.First().Voyage;
+
+        public UnLocode LastUnloadLocation => Legs.Last().UnloadLocation;
+
+        public DateTime FinalArrivalDate => Legs.Last().UnloadTime;
+
+        public Leg NextOf(UnLocode location)
+        {
+            var next = (Leg)null;
+            var currentFound = false;
+
+            using (var en = Legs.GetEnumerator())
+                while (en.MoveNext())
```

Comment:
The extra indentation [shouldn't be necessary with C# 8](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#using-declarations).